editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/DetailsSection.java

```diff
@@ -472,22 +461,25 @@ private IStatus validateMandatoryBehavior(final Parameter prop, IObservableList
         page.layout();
     }
 
-	private boolean shouldHidePropertyFromGroup(final String group, Parameter p, String currentPropertyGroup) {
+	boolean shouldHidePropertyFromGroup(final String group, Parameter p, String currentPropertyGroup) {
 		return isNotMatchingGroup(group, currentPropertyGroup)
 				|| isInternalElementToHide(p)
 				|| isClassParamToHide(p);
 	}
 
 	private boolean isNotMatchingGroup(final String group, String currentPropertyGroup) {
 		if(DEFAULT_GROUP.equals(group)){
-			return currentPropertyGroup != null && currentPropertyGroup.trim().length()>0;
+			return currentPropertyGroup != null && !currentPropertyGroup.trim().isEmpty();
 		} else {
 			return !group.equals(currentPropertyGroup);
 		}
 	}
 
 	private boolean isClassParamToHide(Parameter p) {
-		return CamelComponentUtils.isClassProperty(p) && AbstractCamelModelElement.NODE_KIND_ELEMENT.equalsIgnoreCase(p.getKind());
+		return CamelComponentUtils.isClassProperty(p)
+				&& AbstractCamelModelElement.NODE_KIND_ELEMENT.equalsIgnoreCase(p.getKind())
+				&& !CamelComponentUtils.isDataFormatProperty(p)
+				&& !CamelComponentUtils.isDescriptionProperty(p);
```

Comment:
no need to check for expression, the kind is never Node, it is always "expression" in this case